webrtc.html

```diff
@@ -1862,6 +1868,39 @@
         </dl>
       </section>
     </section>
+    <section>
+      <h3>Session Negotiation Model</h3>
+      <p>Many changes to the PeerConnection state will require communication with
+        the remote side via the signaling channel. The app can be kept informed
+        as to when it needs to do signaling by listening to the
+        negotiationneeded event.</p>
+      <p>If an operation is performed on a RTCPeerConnection that requires signaling,
+        the connection will be marked as needing negotiation. Examples of such
+        operations include adding, stopping, or removing a track.</p>
+      <p>The negotiation-needed state is cleared when setLocalDescription is
+       called (either for an offer or answer), and the supplied description
+       matches the state of the tracks/datachannels that currenly exist on the
+       RTCPeerConnection. Specifically, this means that all live tracks have
+       an associated section in the local description with their MSID,
+       all stopped tracks have been removed from the local description,
+       and, if any data channels have been created, a data section exists in
+       the local description.</p>
+      <p>Note that setLocalDescription(answer) will clear the
+      negotiation-needed state only if the offer had a corresponding section for
+      all the tracks/datachannels on the answerer side. Otherwise, a new offer by
+      the answerer is still needed, and so the state is not cleared.
+      </p>
+      <p>For simplicity, the negotiation-needed state is only cleared by
+       setLocalDescription. It is far better to fire too many negotiationneeded
+       events than too few; spurious negotiationneededs are essentially harmless,
+       whereas missing events may prevent proper operation of the application.</p>
```

Comment:
I agree with Stefan, removing the paragraph would avoid confusion.  This is only design detail (and arguably incorrect, as this design ensures fewer events than might otherwise be generated).